lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -2727,6 +2727,11 @@ void ByteCodeGenerator::EmitFunctionBody(FuncInfo *funcInfo)
     }
     Assert(!pnode->CapturesSyms());
     EmitTopLevelStatement(pnode, funcInfo, false);
+
+    if (funcInfo->IsClassConstructor())
+    {
+        funcInfo->GetParsedFunctionBody()->SetHasNoExplicitReturnValue(false);
```

Comment:
Is this the right place to set this flag?  Seems like this is something that could be set earlier when the function body object is created.